src/EditorFeatures/CSharpTest/SplitComment/SplitCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,268 @@
+                    var history = undoHistoryRegistry.GetHistory(document.GetTextBuffer());
+                    history.Undo(count: originalSelections.Count);
+
+                    var currentSnapshot = document.GetTextBuffer().CurrentSnapshot;
+                    Assert.Equal(originalSnapshot.GetText(), currentSnapshot.GetText());
+                }
+            }
+        }
+
+        /// <summary>
+        /// verifyUndo is needed because of https://github.com/dotnet/roslyn/issues/28033
+        /// Most tests will continue to verifyUndo, but select tests will skip it due to
+        /// this known test infrastructure issure. This bug does not represent a product
+        /// failure.
+        /// </summary>
+        private void TestHandled(
+            string inputMarkup, string expectedOutputMarkup,
+            bool verifyUndo = true, IndentStyle indentStyle = IndentStyle.Smart)
+        {
+            TestWorker(
+                inputMarkup, expectedOutputMarkup,
+                callback: () =>
+                {
+                    Assert.True(false, "Should not reach here.");
+                },
+                verifyUndo, indentStyle);
+        }
+
+        private void TestNotHandled(string inputMarkup)
+        {
+            var notHandled = false;
+            TestWorker(
+                inputMarkup, null,
+                callback: () =>
+                {
+                    notHandled = true;
+                });
+
+            Assert.True(notHandled);
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitStartOfComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //[||] Test Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //
+        // Test Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitMiddleOfComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test [||]Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test 
+        //Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitEndOfComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test Comment[||] 
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test Comment
+        // 
```

Comment:
Given that the space-after-`//` behavior I describe is not implemented (as seen in `TestSplitMiddleOfComment`), I find it strange that you *do* end up with a space after `//` on the second line here.

Reply:
Actually, I don't think this should happen at all, at least not without an option in code style settings to control it. Generally after writing a single line comment and pressing Enter at the end of the line, I want to get back to typing code on the next line, and not have to delete the `//` first.